corehq/form_processor/models/forms.py

```diff
@@ -718,6 +727,15 @@ def unarchive(self, user_id=None, trigger_signals=True):
         if self.is_archived:
             type(self).objects.do_archive(self, False, user_id, trigger_signals)
 
+    def delete(self, leave_tombstone=False):
```

Comment:
Same as cases, default to `leave_tombstone=True`?

Reply:
Just made the change to make leaving a tombstone the default but I wonder if there are existing places where system cases/forms are deleted individually? Nothing really turned up when I looked and I assume most of these places probably use bulk deletion anyways but do you have any ideas about whether these places might exist and if I should specify `leave_tombstone=False` to try and keep DeletedSQLDocs (at least for forms and cases) a log only for user driven deletion?

Reply:
> do you have any ideas about whether these places might exist

I think we always want to leave a tombstone if a form or case is being deleted when the domain is not being deleted.

> [should I] specify `leave_tombstone=False` to try and keep DeletedSQLDocs (at least for forms and cases) a log only for user driven deletion?

Interesting. I suppose it would make sense to specify `leave_tombstone=False` in places where system cases are automatically deleted by the system. For example, @gherceg found a place in CommTrack where something like  a _user owner mapping case_ was being deleted prior to linking a new location. Graham should be able to give more details if you can't find the relevant code.

Aside: what happens to the tombstone if a case is deleted, and then later another case is created with the same case id? Does the tombstone get deleted when the new case is created?

Reply:
> Aside: what happens to the tombstone if a case is deleted, and then later another case is created with the same case id? Does the tombstone get deleted when the new case is created?

hmm I definitely overlooked this scenario and in the current state, it could be problematic if a user tries to delete the second case with the same case_id later on, since it'll pass on creating it, thinking it already exists. How likely is this scenario? Would it help to add `deleted_on` to the unique constraint that already exists for doc_id and object_class_path? or maybe just get rid of that constraint altogether?

Reply:
> How likely is this scenario?

I don't know, but I don't think it will be hard to handle, so we should do that (unless it turns out I am wrong and it's hard).

> Would it help to add `deleted_on` to the unique constraint that already exists for `doc_id` and `object_class_path`? or maybe just get rid of that constraint altogether? or maybe just get rid of that constraint altogether?

Removing that constraint is an intriguing idea. It's useful to guarantee data integrity, but maybe we don't care about duplicates? Another option would be to use an "upsert" operation when writing to the table, which would update rather than insert if a matching record already exists.